### PR TITLE
Prometheus/2022 05 17/use get prefix for metric names

### PR DESCRIPTION
**Pull request by Xaelias**
## 💸 TL;DR
Follow up to https://github.com/reddit/baseplate.py/pull/672
We decided to use static variables / methods for promtheus_metrics.py and standardized how names are generated, as well as how variables themselves are called between http/thrift client/server.

## 🧪 Testing Steps / Validation
tests still pass
This is exclusively refactoring, moving code around and making some things "static". No logic was touched.

## ✅ Checks

- [x] CI tests (if present) are passing
- [x] Adheres to code style for repo
- [x] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

File: baseplate/lib/prometheus_metrics.py [hunk 1]
```diff
@@ -26,316 +26,348 @@
...
-    def get_latency_seconds_metric(self) -> Histogram:
-        return http_client_latency_seconds
+    @classmethod
+    def get_latency_seconds_metric(cls) -> Histogram:
+        return cls.latency_seconds
 
-    def get_requests_total_metric(self) -> Counter:
-        return http_client_requests_total
+    @classmethod
+    def get_requests_total_metric(cls) -> Counter:
```
Comment (nsheaps) [inline baseplate/lib/prometheus_metrics.py:103]: Won't this share the gauge and counter across spans if they're class methods?

Comment (Xaelias) [reply baseplate/lib/prometheus_metrics.py:103]: They are shared. The tags is the only thing differencing these across spans. The underlying object is the same and a singleton. We only ever created one of them outside of the class to begin with.

Comment (nsheaps) [reply baseplate/lib/prometheus_metrics.py:103]: doh, thank you for correcting my line of thought

File: baseplate/lib/prometheus_metrics.py [hunk 2]
```diff
@@ -26,316 +26,348 @@
...
-
-# Gauge showing current number of active requests by a given client
-http_client_active_requests = Gauge(
-    "http_client_active_requests",
-    "Number of active requests for a given client",
-    http_client_active_requests_labels,
-)
-
 
 class PrometheusHTTPClientMetrics:
```
Comment (krav) [inline baseplate/lib/prometheus_metrics.py:30]: Very pro moving these into the class! 

Are these also instantiated when the file is loaded, or when the class is first used? Worried about metrics disappearing on restart.

Comment (Xaelias) [reply baseplate/lib/prometheus_metrics.py:30]: they're static variables, they exist even if the class is never instantiated
I guess I don't know for sure if they exist before they're accessed?
but they wouldn't exist without tags provided either way so I don't think we can lose anything here

Comment (krav) [reply baseplate/lib/prometheus_metrics.py:30]: Ah, I didn't realize but of course they need labels first. Thanks!

**Review by nsheaps — APPROVED** [review]
I like this much better, thanks for doing this!

**Pull request merged**